pageserver/src/import_datadir.rs

```diff
@@ -216,7 +223,19 @@ fn import_rel<R: Repository, Reader: Read>(
             Err(err) => match err.kind() {
                 std::io::ErrorKind::UnexpectedEof => {
                     // reached EOF. That's expected.
-                    ensure!(blknum == nblocks as u32, "unexpected EOF");
+                    let relative_blknum =
+                        blknum - segno * (1024 * 1024 * 1024 / pg_constants::BLCKSZ as u32);
+
+                    // Update relation size
+                    // This update is important,
+                    // because rel_creation above could have set
+                    // incorrect relsize for multi-segment relations
+                    //
+                    // If we process rel segments out of order,
+                    // put_rel_extend will skip the update.
+                    modification.put_rel_extend(rel, blknum)?;
```

Comment:
Took me a while to understand this. It might be more readable if moved after the for loop. That way the function will have 3 parts:
1. Create relation if not exists
2. Put all the images
3. Extend size

Where 2 and 3 are unrelated